src/realm/sync/transform.cpp

```diff
@@ -1597,8 +1597,10 @@ DEFINE_MERGE(Instruction::EraseColumn, Instruction::Update)
 DEFINE_MERGE(Instruction::ArrayInsert, Instruction::Update)
 {
     if (same_container(left, right)) {
-        REALM_ASSERT(right.is_array_update());
+        REALM_MERGE_ASSERT(right.is_array_update());
```

Comment:
Out of curiosity what the difference between REALM_ASSERT and REALM_MERGE_ASSERT()?

Reply:
REALM_ASSERT crashes the process, REALM_MERGE_ASSERT rejects the changeset as invalid and carries on.